src/mpid/ch4/netmod/ofi/util.c

```diff
@@ -501,9 +501,11 @@ static inline void create_dt_map()
 
 static inline void add_index(MPI_Datatype datatype, int *idx)
 {
-    MPIR_Datatype *dt_ptr;
-    MPIR_Datatype_get_ptr(datatype, dt_ptr);
-    MPIDI_OFI_DATATYPE(dt_ptr).index = *idx;
+    if (datatype != MPI_DATATYPE_NULL) {
```

Comment:
I would add a comment here that some datatypes may not be supported depending on the underlying platform, in which case `datatype` becomes `MPI_DATATYPE_NULL`.

Reply:
OK, will add.